astropy/time/core.py

```diff
@@ -408,6 +409,40 @@ def now(cls):
 
     info = TimeInfo()
 
+    @classmethod
+    def strptime(cls, time_string, format_string):
+        """
+        Parse a string to a Time according to a format specification.
+        See time.strptime documentation for format specification.
+
+        >>> Time.strptime('2012-06-30T23:59:60', '%Y-%m-%dT%H:%M:%S')
+        <Time object: scale='utc' format='isot' value=2012-06-30T23:59:60.000>
+
+        Parameters
+        ----------
+        time_string : string, sequence, ndarray
+            Objects containing time data of type string
+        format_string : string
+            String specifying format of time_string.
+
+        Returns
+        -------
+        astropy.time.Time
+            A new `Time` object corresponding to the input time_string.
+
+        """
+        time_array = np.array(time_string)
+        to_string = (str if time_array.dtype.kind == 'U' else
+                     lambda x: str(x.item(), encoding='ascii'))
+        iterator = np.nditer([time_array, None],
+                             op_dtypes=[time_array.dtype, '<U30'])
+
+        for v, s in iterator:
+            time_tuple = strptime(to_string(v), format_string)
+            s[...] = strftime('%Y-%m-%dT%H:%M:%S', time_tuple)
```

Comment:
If you have the `time_tuple` can you just use standard Python string formatting?  Might this be faster?